libs/orders/src/lib/order-hooks/use-order-edit.tsx

```diff
@@ -0,0 +1,126 @@
+  const [updatedOrder, setUpdatedOrder] =
+    useState<OrderEvent_busEvents_event_Order | null>(null);
+  const client = useApolloClient();
+  const subRef = useRef<Subscription | null>(null);
+
+  const reset = useCallback(() => {
+    resetTransaction();
+    setUpdatedOrder(null);
+    subRef.current?.unsubscribe();
+  }, [resetTransaction]);
+
+  useEffect(() => {
+    return () => {
+      resetTransaction();
+      setUpdatedOrder(null);
+      subRef.current?.unsubscribe();
+    };
+  }, [resetTransaction]);
+
+  const edit = useCallback(
+    async (order) => {
+      if (!keypair) {
+        return;
+      }
+
+      setUpdatedOrder(null);
+
+      try {
+        const res = await send({
+          pubKey: keypair.pub,
+          propagate: true,
+          orderAmendment: {
+            orderId: order.id,
+            marketId: order?.market?.id,
+            price: {
+              value: removeDecimal(
+                order.price,
+                order.market.decimalPlaces ?? 0
+              ),
+            },
+            timeInForce:
+              `TIME_IN_FORCE_${order.timeInForce}` as VegaWalletOrderTimeInForce,
+            sizeDelta: 0,
+            expiresAt: order.expiresAt
+              ? {
+                  value:
+                    // Wallet expects timestamp in nanoseconds,
+                    // we don't have that level of accuracy so just append 6 zeroes
+                    new Date(order.expiresAt).getTime().toString() + '000000',
+                }
+              : undefined,
+          } as unknown as OrderAmendmentBodyOrderAmendment,
+        } as OrderAmendmentBody);
+
+        if (res?.signature) {
+          const resId = order.id ?? determineId(res.signature);
+          setUpdatedOrder(null);
+          // setId(resId);
+          if (resId) {
+            // Start a subscription looking for the newly created order
+            subRef.current = client
+              .subscribe<OrderEvent, OrderEventVariables>({
+                query: ORDER_EVENT_SUB,
+                variables: { partyId: keypair?.pub || '' },
+              })
+              .subscribe(({ data }) => {
+                if (!data?.busEvents?.length) {
+                  return;
+                }
+
+                // No types available for the subscription result
+                const matchingOrderEvent = data.busEvents.find((e) => {
+                  if (e.event.__typename !== 'Order') {
+                    return false;
+                  }
+
+                  return e.event.id === resId;
+                });
+
+                if (
+                  matchingOrderEvent &&
+                  matchingOrderEvent.event.__typename === 'Order'
+                ) {
+                  setUpdatedOrder(matchingOrderEvent.event);
+                  subRef.current?.unsubscribe();
+                }
+              });
+          }
+        }
+        return res;
+      } catch (e) {
+        Sentry.captureException(e);
```

Comment:
question (non-blocking): not sure what happens if the app doesn't have a Sentry handler and tried using this hook. for me it feels like this should be handled on the app level, and just pass in an onQueryError attribute to the hook.

Reply:
So I think implementing https://github.com/vegaprotocol/frontend-monorepo/issues/644 should handle this